### PR TITLE
Reduce amount of / consolidate config files

### PR DESCRIPTION
**Pull request by Yannic92**
Before this PR we supported the following different "hosting environments":
- docker
- cloud
- filebased
- dev
Which expected config in different files. For example: The docker hosting environment expected the config to be present in `servicename-docker.conf`

This lead to confusion and duplication in configuration files. This PR now only supports the following "hosting environments":

- docker (just uses the default servicename.conf)
- filebased (uses a configurable file as config)
- dev (uses the servicename-dev.conf)

So no `-cloud` and no `-docker` files are required anymore.

VCAP and docker secret support is now available in all kind of config files not only in the `-cloud` config file.

### CODE REVIEW

File: services/gateway/starter/src/main/resources/gateway.conf [hunk 1]
```diff
@@ -264,21 +265,22 @@ ditto {
...
 
-      // how long to wait for messages from other cluster members
+      # how long to wait for messages from other cluster members
       ask-timeout = 5s
+      ask-timeout = ${?STATISTICS_UPDATE_INTERVAL}
 
-      // how often to update hot entities count for the public
-      update-interval = 30s
+      # how often to update hot entities count for the public
+      update-interval = 15s
```
Comment (jbartelh) [inline services/gateway/starter/src/main/resources/gateway.conf:273]: Is this a breaking change in ditto?

File: services/gateway/starter/src/main/resources/gateway.conf [hunk 2]
```diff
@@ -264,21 +265,22 @@ ditto {
...
       update-interval = ${?STATISTICS_UPDATE_INTERVAL}
 
-      // minimum pause between computations of statistics details
-      details-expire-after = 5s
+      # minimum pause between computations of statistics details
+      details-expire-after = 3s
```
Comment (jbartelh) [inline services/gateway/starter/src/main/resources/gateway.conf:277]: Is this a breaking change in ditto?

File: services/gateway/starter/src/main/resources/gateway.conf [hunk 3]
```diff
@@ -372,7 +379,7 @@ akka {
         #
         # `relaxed`: all visible 7-Bit ASCII chars are allowed
         #
-        uri-parsing-mode = relaxed
+        uri-parsing-mode = strict
```
Comment (jbartelh) [inline services/gateway/starter/src/main/resources/gateway.conf:382]: Is this a breaking change in ditto?

**Review by jbartelh — APPROVED** [review]
Not sure if the three mentioned config value changes are categorized as 'breaking changes' or not, but we should at least mention them in the upcoming release notes so that ditto user can adapt if needed. What do you say @thjaeckle ?

The rest LGTM

File: services/gateway/starter/src/main/resources/gateway.conf (hunk 1 above)
Comment (thjaeckle) [reply services/gateway/starter/src/main/resources/gateway.conf:273]: no

File: services/gateway/starter/src/main/resources/gateway.conf (hunk 2 above)
Comment (thjaeckle) [reply services/gateway/starter/src/main/resources/gateway.conf:277]: no

File: services/gateway/starter/src/main/resources/gateway.conf (hunk 3 above)
Comment (thjaeckle) [reply services/gateway/starter/src/main/resources/gateway.conf:382]: hmmh, I think we once configured `strict` for a very good reason. (that I can't think of any more)
why was this changed to `relaxed` again?

Comment (Yannic92) [reply services/gateway/starter/src/main/resources/gateway.conf:382]: I think you're right. Let's revert this change and set it to relaxed again.

**Review by thjaeckle — APPROVED** [review]

**Pull request merged**